app/models/glitch/keyword_mute.rb

```diff
@@ -33,68 +33,74 @@ def invalidate_cached_matchers
     Rails.cache.delete(TagMatcher.cache_key(account_id))
   end
 
-  class RegexpMatcher
+  class CachedKeywordMute
+    attr_reader :keyword
+    attr_reader :whole_word
+
+    def initialize(keyword, whole_word)
+      @keyword = keyword
+      @whole_word = whole_word
+    end
+
+    def boundary_regex_for_keyword
+      sb = keyword =~ /\A[[:word:]]/ ? '\b' : ''
+      eb = keyword =~ /[[:word:]]\Z/ ? '\b' : ''
+
+      /(?mix:#{sb}#{Regexp.escape(keyword)}#{eb})/
+    end
+
+    def matches?(str)
+      str =~ (whole_word ? boundary_regex_for_keyword : /#{keyword}/i)
```

Comment:
I would like to get an understanding of what the memory impact (e.g. objects created, GC pressure) is from all of these temporary regex objects.  Is there a way to do this that can provide more direct answers than poking at ObjectSpace and GC stats? It's been a while since I dipped into Ruby memory profiling and am currently catching up.